From dff9359daac153961ffcb6b719c221265af33a04 Mon Sep 17 00:00:00 2001
From: Fangrui Song <i@maskray.me>
Date: Sun, 18 Jun 2023 19:30:16 -0700
Subject: [PATCH] [XRay] Mark Mach-O xray_instr_map and xray_fn_idx as
 S_ATTR_LIVE_SUPPORT

Add the `S_ATTR_LIVE_SUPPORT` attribute to the sections so that `ld -dead_strip`
will retain subsections that reference live functions, once we we add linker
private "l" symbols as atoms.
---
 llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp                | 6 ++++--
 .../CodeGen/AArch64/xray-attribute-instrumentation.ll     | 6 +++---
 llvm/test/CodeGen/AArch64/xray-omit-function-index.ll     | 2 +-
 .../AArch64/xray-partial-instrumentation-skip-entry.ll    | 2 +-
 .../AArch64/xray-partial-instrumentation-skip-exit.ll     | 2 +-
 llvm/test/CodeGen/AArch64/xray-tail-call-sled.ll          | 8 ++++----
 .../CodeGen/ARM/xray-armv6-attribute-instrumentation.ll   | 4 ++--
 .../CodeGen/ARM/xray-armv7-attribute-instrumentation.ll   | 4 ++--
 llvm/test/CodeGen/X86/xray-attribute-instrumentation.ll   | 8 ++++----
 .../X86/xray-partial-instrumentation-skip-entry.ll        | 8 ++++----
 .../CodeGen/X86/xray-partial-instrumentation-skip-exit.ll | 8 ++++----
 llvm/test/CodeGen/X86/xray-tail-call-sled.ll              | 8 ++++----
 12 files changed, 34 insertions(+), 32 deletions(-)

diff --git a/llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp b/llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp
index a10b5e57b816..88cce977a347 100644
--- a/llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp
+++ b/llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp
@@ -3386,11 +3386,13 @@ void AsmPrinter::emitXRayTable() {
           "xray_fn_idx", ELF::SHT_PROGBITS, Flags | ELF::SHF_WRITE, 0,
           GroupName, MCSection::NonUniqueID, LinkedToSym);
   } else if (MF->getSubtarget().getTargetTriple().isOSBinFormatMachO()) {
-    InstMap = OutContext.getMachOSection("__DATA", "xray_instr_map", 0,
+    InstMap = OutContext.getMachOSection("__DATA", "xray_instr_map",
+                                         MachO::S_ATTR_LIVE_SUPPORT,
                                          SectionKind::getReadOnlyWithRel());
     if (TM.Options.XRayFunctionIndex)
       FnSledIndex = OutContext.getMachOSection(
-          "__DATA", "xray_fn_idx", 0, SectionKind::getReadOnlyWithRel());
+          "__DATA", "xray_fn_idx", MachO::S_ATTR_LIVE_SUPPORT,
+          SectionKind::getReadOnlyWithRel());
   } else {
     llvm_unreachable("Unsupported target");
   }
diff --git a/llvm/test/CodeGen/AArch64/xray-attribute-instrumentation.ll b/llvm/test/CodeGen/AArch64/xray-attribute-instrumentation.ll
index 6c269869f11c..f196a5dc89ba 100644
--- a/llvm/test/CodeGen/AArch64/xray-attribute-instrumentation.ll
+++ b/llvm/test/CodeGen/AArch64/xray-attribute-instrumentation.ll
@@ -21,7 +21,7 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .xword .Lxray_sled_1
 ; CHECK-LINUX-LABEL: Lxray_sleds_end0:
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS:         .quad Lxray_sled_1
@@ -47,7 +47,7 @@ define i32 @bar() nounwind noinline uwtable "function-instrument"="xray-never" "
 ; CHECK-LINUX:         .xword .Lxray_sled_3
 ; CHECK-LINUX-LABEL: Lxray_sleds_end1:
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start1:
 ; CHECK-MACOS:         .quad Lxray_sled_2
 ; CHECK-MACOS:         .quad Lxray_sled_3
@@ -73,7 +73,7 @@ define i32 @instrumented() nounwind noinline uwtable "xray-instruction-threshold
 ; CHECK-LINUX:         .xword .Lxray_sled_5
 ; CHECK-LINUX-LABEL: Lxray_sleds_end2:
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start2:
 ; CHECK-MACOS:         .quad Lxray_sled_4
 ; CHECK-MACOS:         .quad Lxray_sled_5
diff --git a/llvm/test/CodeGen/AArch64/xray-omit-function-index.ll b/llvm/test/CodeGen/AArch64/xray-omit-function-index.ll
index bc450d726202..36faabe57ddc 100644
--- a/llvm/test/CodeGen/AArch64/xray-omit-function-index.ll
+++ b/llvm/test/CodeGen/AArch64/xray-omit-function-index.ll
@@ -31,7 +31,7 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .xword .Lxray_sled_1
 ; CHECK-LINUX-LABEL: Lxray_sleds_end0:
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS:         .quad Lxray_sled_1
diff --git a/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-entry.ll b/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-entry.ll
index 00055b5f439e..ed2d488ed010 100644
--- a/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-entry.ll
+++ b/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-entry.ll
@@ -22,7 +22,7 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .xword .Lxray_sled_0
 ; CHECK-LINUX-LABEL: Lxray_sleds_end0:
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS-LABEL: Lxray_sleds_end0:
diff --git a/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-exit.ll b/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-exit.ll
index 2097be44d16b..1908a4e3ab58 100644
--- a/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-exit.ll
+++ b/llvm/test/CodeGen/AArch64/xray-partial-instrumentation-skip-exit.ll
@@ -22,7 +22,7 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .xword .Lxray_sled_0
 ; CHECK-LINUX-LABEL: Lxray_sleds_end0:
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS-LABEL: Lxray_sleds_end0:
diff --git a/llvm/test/CodeGen/AArch64/xray-tail-call-sled.ll b/llvm/test/CodeGen/AArch64/xray-tail-call-sled.ll
index 67de3246c035..aa3802bfd249 100644
--- a/llvm/test/CodeGen/AArch64/xray-tail-call-sled.ll
+++ b/llvm/test/CodeGen/AArch64/xray-tail-call-sled.ll
@@ -28,14 +28,14 @@ define i32 @callee() nounwind noinline uwtable "function-instrument"="xray-alway
 ; CHECK-LINUX:         .xword .Lxray_sleds_start0
 ; CHECK-LINUX-NEXT:    .xword .Lxray_sleds_end0
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS-NEXT:  [[TMP:Ltmp[0-9]+]]:
 ; CHECK-MACOS:         .quad Lxray_sled_0-[[TMP]]
 ; CHECK-MACOS:       [[TMP:Ltmp[0-9]+]]:
 ; CHECK-MACOS-NEXT:    .quad Lxray_sled_1-[[TMP]]
 ; CHECK-MACOS-LABEL: Lxray_sleds_end0:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start0
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end0
 
@@ -65,11 +65,11 @@ define i32 @caller() nounwind noinline uwtable "function-instrument"="xray-alway
 ; CHECK-LINUX:         .xword .Lxray_sleds_start1
 ; CHECK-LINUX-NEXT:    .xword .Lxray_sleds_end1
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start1:
 ; CHECK-MACOS:         .quad Lxray_sled_2
 ; CHECK-MACOS:         .quad Lxray_sled_3
 ; CHECK-MACOS-LABEL: Lxray_sleds_end1:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start1
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end1
diff --git a/llvm/test/CodeGen/ARM/xray-armv6-attribute-instrumentation.ll b/llvm/test/CodeGen/ARM/xray-armv6-attribute-instrumentation.ll
index b37b22b5c359..965e1781884d 100644
--- a/llvm/test/CodeGen/ARM/xray-armv6-attribute-instrumentation.ll
+++ b/llvm/test/CodeGen/ARM/xray-armv6-attribute-instrumentation.ll
@@ -33,11 +33,11 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .long .Lxray_sleds_start0
 ; CHECK-LINUX-NEXT:    .long .Lxray_sleds_end0
 
-; CHECK-IOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-IOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-IOS-LABEL: Lxray_sleds_start0:
 ; CHECK-IOS:         .long Lxray_sled_0
 ; CHECK-IOS:         .long Lxray_sled_1
 ; CHECK-IOS-LABEL: Lxray_sleds_end0:
-; CHECK-IOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-IOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-IOS:         .long Lxray_sleds_start0
 ; CHECK-IOS-NEXT:    .long Lxray_sleds_end0
diff --git a/llvm/test/CodeGen/ARM/xray-armv7-attribute-instrumentation.ll b/llvm/test/CodeGen/ARM/xray-armv7-attribute-instrumentation.ll
index c3be28828be2..1570a8718194 100644
--- a/llvm/test/CodeGen/ARM/xray-armv7-attribute-instrumentation.ll
+++ b/llvm/test/CodeGen/ARM/xray-armv7-attribute-instrumentation.ll
@@ -23,11 +23,11 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .long .Lxray_sleds_start0
 ; CHECK-LINUX-NEXT:    .long .Lxray_sleds_end0
 
-; CHECK-IOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-IOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-IOS-LABEL: Lxray_sleds_start0:
 ; CHECK-IOS:         .long Lxray_sled_0
 ; CHECK-IOS:         .long Lxray_sled_1
 ; CHECK-IOS-LABEL: Lxray_sleds_end0:
-; CHECK-IOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-IOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-IOS:         .long Lxray_sleds_start0
 ; CHECK-IOS-NEXT:    .long Lxray_sleds_end0
diff --git a/llvm/test/CodeGen/X86/xray-attribute-instrumentation.ll b/llvm/test/CodeGen/X86/xray-attribute-instrumentation.ll
index f7137f50f49c..dad876dfcaca 100644
--- a/llvm/test/CodeGen/X86/xray-attribute-instrumentation.ll
+++ b/llvm/test/CodeGen/X86/xray-attribute-instrumentation.ll
@@ -23,12 +23,12 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .quad .Lxray_sleds_start0
 ; CHECK-LINUX-NEXT:    .quad .Lxray_sleds_end0
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS:         .quad Lxray_sled_1
 ; CHECK-MACOS-LABEL: Lxray_sleds_end0:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start0
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end0
 
@@ -70,7 +70,7 @@ NotEqual:
 ; CHECK-LINUX:         .quad .Lxray_sleds_start1
 ; CHECK-LINUX-NEXT:    .quad .Lxray_sleds_end1
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start1:
 ; CHECK-MACOS:       [[TMP:Ltmp[0-9]+]]:
 ; CHECK-MACOS-NEXT:    .quad Lxray_sled_2-[[TMP]]
@@ -79,6 +79,6 @@ NotEqual:
 ; CHECK-MACOS:       [[TMP:Ltmp[0-9]+]]:
 ; CHECK-MACOS-NEXT:    .quad Lxray_sled_4-[[TMP]]
 ; CHECK-MACOS-LABEL: Lxray_sleds_end1:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start1
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end1
diff --git a/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-entry.ll b/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-entry.ll
index 465aab3e32b3..839afb682d57 100644
--- a/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-entry.ll
+++ b/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-entry.ll
@@ -19,11 +19,11 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .quad .Lxray_sleds_start0
 ; CHECK-LINUX-NEXT:    .quad .Lxray_sleds_end0
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS-LABEL: Lxray_sleds_end0:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start0
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end0
 
@@ -58,11 +58,11 @@ NotEqual:
 ; CHECK-LINUX:         .quad .Lxray_sleds_start1
 ; CHECK-LINUX-NEXT:    .quad .Lxray_sleds_end1
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start1:
 ; CHECK-MACOS:         .quad Lxray_sled_1
 ; CHECK-MACOS:         .quad Lxray_sled_2
 ; CHECK-MACOS-LABEL: Lxray_sleds_end1:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start1
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end1
diff --git a/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-exit.ll b/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-exit.ll
index 7d9ec92d770e..a2229c7d78f8 100644
--- a/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-exit.ll
+++ b/llvm/test/CodeGen/X86/xray-partial-instrumentation-skip-exit.ll
@@ -21,11 +21,11 @@ define i32 @foo() nounwind noinline uwtable "function-instrument"="xray-always"
 ; CHECK-LINUX:         .quad .Lxray_sleds_start0
 ; CHECK-LINUX-NEXT:    .quad .Lxray_sleds_end0
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS-LABEL: Lxray_sleds_end0:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start0
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end0
 
@@ -58,10 +58,10 @@ NotEqual:
 ; CHECK-LINUX:         .quad .Lxray_sleds_start1
 ; CHECK-LINUX-NEXT:    .quad .Lxray_sleds_end1
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start1:
 ; CHECK-MACOS:         .quad Lxray_sled_1
 ; CHECK-MACOS-LABEL: Lxray_sleds_end1:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start1
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end1
diff --git a/llvm/test/CodeGen/X86/xray-tail-call-sled.ll b/llvm/test/CodeGen/X86/xray-tail-call-sled.ll
index d63102696a6c..b755f610ccd6 100644
--- a/llvm/test/CodeGen/X86/xray-tail-call-sled.ll
+++ b/llvm/test/CodeGen/X86/xray-tail-call-sled.ll
@@ -22,12 +22,12 @@ define dso_local i32 @callee() nounwind noinline uwtable "function-instrument"="
 ; CHECK-LINUX:         .quad .Lxray_sleds_start0
 ; CHECK-LINUX-NEXT:    .quad .Lxray_sleds_end0
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start0:
 ; CHECK-MACOS:         .quad Lxray_sled_0
 ; CHECK-MACOS:         .quad Lxray_sled_1
 ; CHECK-MACOS-LABEL: Lxray_sleds_end0:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start0
 ; CHECK-MACOS-NEXT:    .quad Lxray_sleds_end0
 
@@ -54,11 +54,11 @@ define dso_local i32 @caller() nounwind noinline uwtable "function-instrument"="
 ; CHECK-LINUX:         .quad .Lxray_sleds_start1
 ; CHECK-LINUX:         .quad .Lxray_sleds_end1
 
-; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_instr_map,regular,live_support{{$}}
 ; CHECK-MACOS-LABEL: Lxray_sleds_start1:
 ; CHECK-MACOS:         .quad Lxray_sled_2
 ; CHECK-MACOS:         .quad Lxray_sled_3
 ; CHECK-MACOS-LABEL: Lxray_sleds_end1:
-; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx{{$}}
+; CHECK-MACOS-LABEL: .section __DATA,xray_fn_idx,regular,live_support{{$}}
 ; CHECK-MACOS:         .quad Lxray_sleds_start1
 ; CHECK-MACOS:         .quad Lxray_sleds_end1
-- 
2.39.3

